update comparison in readme

antonk52 opened the pull request:
Hi, I've come across your example and I believe there are a few things we can improve around webpack specifically and overall. 

- we can add `loose: true` to a `@babel/preset-env` option, which slightly reduces the build size
- if we have a speedy build for webpack we do not need the `--progress` flag, it also increases the build time 0.5/0.7s
- we can add a `package-lock.json` file or specify the bundlers versions directly. Below are my results for webpack 4.27.1, Parcel 1.10.3 and Rollup 0.63.4

As a side note which is not included in this PR is the incremental build for `webpack:serve` can be reduced down to ~400ms if set up right `cache-loader`. I was not sure wether this repo is more of a comparison for a zero-config setups or how well we can configure each in order to see the performance difference. I will be happy to improve webpack build times as an addition to this PR or do it in anothher PR.

Either way thank you for the repo!

mkalygin commented:
@antonk52 awesome, thanks! From the first glance it looks good. 👍I'll review it this week.

My initial intention was to do as less config as possible in order to get it up and running. And also compare my first impressions. The impressions are biased because I didn't give much love to Rollup since I'm noob in it. I'm not very proficient with Webpack optimizations either, so any PRs with good practices are welcome! I'd probably make another branch with optimized setups for this, but overall I'd like the idea to compare solid production ready setups for all the bundlers.

antonk52 commented:
Sounds great. I was going to propose to perhaps have a separate config with a more precise(complex) webpack set up which is more efficient but requires more tinkering. What do you think? The same can apply to both Parcel and Rollup. I have not had a proper experience with either of them. Maybe more experienced devs can fill in on those.

mkalygin commented:
@antonk52 I love the idea. We can make new scripts for it. Something like `advanced:webpack:build`, etc. Parcel is probably is not a player here since it's a zero config bundler, it's very inflexible in this matter. Rollup will probably shine here, but will require more expertise than I currently have, so maybe someone else will be able to provide a good setup for it. The problem with Rollup is that it's hard to integrate into existing project and it requires a lot of manual setup of the libraries. But it may give us a very slim bundle.

antonk52 commented:
Sorry for a delay. I've added the advanced webpack config which requires two extra dependencies(noted in the readme). I am pretty happy with the numbers so far. I believe that's a good idea if you can run the tests on your machine(the one from the readme) to compare the speed for each bundler since there was not much info in the readme how these tests are performed and I have a different laptop.

A little note on what have I done. I noticed that you used `devtool: 'source-map'` which is pretty expensive during developments and it's a good practice to inline source maps in dev mode. I did not edit the base webpack config. Yet in advanced section the `devtool` is set to `inline-eval` which is a much cheaper version that gave the largest speed boost. You can also set it to `inline-source-map` which will provide better source maps by appending a flag to the command like so
```
npm run webpack-advanced:serve --sourcemaps
```
I've also added `cache-loader` with specified directory for caching as well as enabled caching for `babel-loader`. The last thing is `thread-loader` which helps to parallel the work done by the expensive loaders.

I am happy to extend the readme or answer any questions.

Hope this helps.

mkalygin commented:
@antonk52 thanks for your work! I appreciate it. I'm sorry, I haven't got a chance to look at it yet, but I'm going to do so this weekend.

I think I'll just merge your work and update README tables. I also wanted to provide a script to generate these tables automatically, so everyone could run them on their machines.

antonk52 commented:
Thanks a bunch. There are still other optimisations which can be applied to smooth the workflow for webpack. Some of them are more applicable for larger applications so they simply do not apply here. 

I think it's a great idea to generate tables dynamically yet there is a food for thought that the incremental builds can be different depending which parts of the app you edit. Therefore, those should  be checked separately.

mkalygin commented:
@antonk52 I'm going to run the tests real quick and update the README. I'm still thinking on how to automate them since your point on different results for different file edits is valid.
